src/main/java/org/dcsa/ebl/service/impl/ShippingInstructionServiceImpl.java

```diff
@@ -72,80 +76,212 @@
         .flatMap(
             si -> {
-              if (shippingInstructionTO.getDocumentParties() == null) return Mono.just(si);
-              return documentPartyService
-                  .createDocumentPartiesByShippingInstructionID(
-                      si.getShippingInstructionID(), shippingInstructionTO.getDocumentParties())
-                  .then(Mono.just(si));
+              shippingInstructionRequest.setShippingInstructionID(si.getShippingInstructionID());
+              shippingInstructionRequest.setDocumentStatus(si.getDocumentStatus());
+              shippingInstructionRequest.setShippingInstructionCreatedDateTime(
+                  si.getShippingInstructionCreatedDateTime());
+              shippingInstructionRequest.setShippingInstructionUpdatedDateTime(
+                  si.getShippingInstructionUpdatedDateTime());
+              return Mono.when(
+                      locationService
+                          .resolveLocationByTO(
+                              si.getPlaceOfIssueID(),
+                              shippingInstructionRequest.getPlaceOfIssue(),
+                              placeOfIssue ->
+                                  shippingInstructionRepository.setPlaceOfIssueFor(
+                                      placeOfIssue, si.getShippingInstructionID()))
+                          .doOnNext(shippingInstructionRequest::setPlaceOfIssue),
+                      resolveShipmentEquipmentsForShippingInstructionID(
+                              shippingInstructionRequest.getShipmentEquipments(),
+                              shippingInstructionRequest)
+                          .doOnNext(shippingInstructionRequest::setShipmentEquipments),
+                      resolveDocumentPartiesForShippingInstructionID(
+                              si.getShippingInstructionID(),
+                              shippingInstructionRequest.getDocumentParties())
+                          .doOnNext(shippingInstructionRequest::setDocumentParties),
+                      resolveReferencesForShippingInstructionID(
+                              si.getShippingInstructionID(),
+                              shippingInstructionRequest.getReferences())
+                          .doOnNext(shippingInstructionRequest::setReferences))
+                  .thenReturn(shippingInstructionRequest);
             })
+        .flatMap(createShipmentEventFromDocumentStatus)
         .flatMap(
-            si -> {
-              if (shippingInstructionTO.getReferences() == null) return Mono.just(si);
-              return referenceService
-                  .createReferencesByShippingInstructionIDAndTOs(
-                      si.getShippingInstructionID(), shippingInstructionTO.getReferences())
-                  .then(Mono.just(si));
-            })
+            siTO -> Mono.just(shippingInstructionMapper.dtoToShippingInstructionResponseTO(siTO)));
+  }
+
+  private Mono<List<ReferenceTO>> resolveReferencesForShippingInstructionID(
```

Comment:
use: ReferenceService::resolveReferencesForShippingInstructionID in event_core

Reply:
done